keyboards/contra/keymaps/maxr1998/keymap.c

```diff
@@ -125,20 +146,36 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
         unregister_code(DE_SS);
       }
       return true;
-    case NR_L3:
+    case KC_NR:
       if (record->event.pressed) {
-        register_code(KC_ALGR);
-        layer_on(_NUMROW_L3);
+        layer_on(_NR);
       } else {
-        layer_off(_NUMROW_L3);
-        unregister_code(KC_ALGR);
+        layer_off(_NR);
       }
+      update_tri_layer_user();
+      return false;
+    case KC_N3:
+      if (record->event.pressed) {
+        layer_on(_NR_L3);
+      } else {
+        layer_off(_NR_L3);
+      }
+      update_tri_layer_user();
       return false;
     default:
       return true;
   }
 }
 
+void update_tri_layer_user(void) {
```

Comment:
You could probably do this with `layer_state_set_user`, but this is absolutely fine.

Reply:
True, that would've been possible. But then it would be called on every `layer_on` and `layer_off`, and not just for layers `_NR` and `_NR_L3`, right?